ion/symbols_table_test.go

```diff
@@ -0,0 +1,268 @@
+			localSID: symbolIDSharedTable,
+			Source:   newSource(symbolTextIon, symbolIDSharedTable),
+		},
+	}
+
+	// explicitly make a new system table to test against so we don't manipulate the global one
+	table := newSystemSymbolTable()
+	return symbolTableCase{
+		desc:    "systemSymbolTable",
+		kind:    symbolTableKindSystem,
+		name:    symbolTextIon,
+		version: 1,
+		maxSID:  9,
+		tokens:  tokens,
+		table:   *table,
+	}
+}
+
+func makeSymbolTableCase(
+	desc string,
+	start int64,
+	kind symbolTableKind,
+	name string,
+	version int32,
+	makeToken func(text string, sid int64) SymbolToken,
+	newTable func(imports ...SymbolTable) *SymbolTable,
+	texts []string,
+	imports ...SymbolTable) symbolTableCase {
+
+	// expect that we start at end of system table
+	var sid int64 = start
+	for _, imp := range imports {
+		sid += imp.maxSID
+	}
+
+	// manually calculate what the implementation should do
+	var tokens []SymbolToken
+	for _, text := range texts {
+		sid += 1
+		tokens = append(tokens, makeToken(text, sid))
+	}
+
+	// construct the table to test
+	table := newTable(imports...)
+	for _, text := range texts {
+		table.InternToken(text)
+	}
+
+	return symbolTableCase{
+		desc:    desc,
+		kind:    kind,
+		name:    name,
+		version: version,
+		imports: imports,
+		maxSID:  sid,
+		tokens:  tokens,
+		table:   *table,
+	}
+}
+
+func makeSharedSymbolTableCase(
+	desc string, name string, version int32, texts []string, imports ...SymbolTable) symbolTableCase {
+	makeToken := func(text string, sid int64) SymbolToken {
+		source := ImportSource{
+			Table: name,
+			SID:   sid,
+		}
+		return SymbolToken{
+			Text:     newString(text),
+			localSID: sid,
+			Source:   &source,
+		}
+	}
+	newTable := func(imports ...SymbolTable) *SymbolTable {
+		return newSharedSymbolTable(name, version, imports...)
+	}
+	return makeSymbolTableCase(
+		desc, 0, symbolTableKindShared, name, version, makeToken, newTable, texts, imports...)
+}
+
+func makeLocalSymbolTableCase(desc string, texts []string, imports ...SymbolTable) symbolTableCase {
+	makeToken := func(text string, sid int64) SymbolToken {
+		return SymbolToken{
+			Text:     newString(text),
+			localSID: sid,
+			Source:   nil,
+		}
+	}
+	return makeSymbolTableCase(
+		desc, 9, symbolTableKindLocal, "", 0, makeToken, newLocalSymbolTable, texts, imports...)
+}
+
+// Arbitrary, deterministic seed for our symbol table testing
+const testSystemSymbolTableSeed = 0x03CB815D3119751B
+
+func TestSymbolTable(t *testing.T) {
+	// TODO test tables with imports
+	cases := []symbolTableCase{
+		makeSystemSymbolTableCase(),
+		makeLocalSymbolTableCase("Empty LST", nil),
+		makeLocalSymbolTableCase("Simple LST", []string{"a", "b", "c"}),
+		makeSharedSymbolTableCase("Empty SST", "foo", 1, nil),
+		makeSharedSymbolTableCase("Simple SST", "bar", 3, []string{"cat", "dog", "moose"}),
+	}
+
+	for _, c := range cases {
+		t.Run(c.desc, func(t *testing.T) {
+			if c.kind != c.table.kind {
```

Comment:
You could create a cmp.Transform to turn your SymbolTable into a string instead of making your test all wordy.  Totally a style choice.

Reply:
I will add a comment to consider doing this in the future.